llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -1484,6 +1535,15 @@ void RISCVInsertVSETVLI::doLocalPostpass(MachineBasicBlock &MBB) {
       Used.demandVL();
 
     if (NextMI) {
+      if (auto NewVType = canAdjustSEWLMULRatio(MI, *NextMI, Used)) {
+        MI.getOperand(2).setImm(*NewVType);
```

Comment:
Have we properly guarded for the save/restore form which has a vreg as operand 2?